src/main/java/uk/gov/companieshouse/api/accounts/controller/SmallFullController.java

```diff
@@ -32,4 +33,14 @@ public ResponseEntity create(@Valid @RequestBody SmallFull smallFull,
         SmallFull result = smallFullService.save(smallFull, transaction.getCompanyNumber());
         return ResponseEntity.status(HttpStatus.CREATED).body(result);
     }
+
+    @GetMapping(value = "/transactions/{transactionId}/company-accounts/{companyAccountId}/small-full",
+        produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity getCompanyAccount(HttpServletRequest request) {
+
+        SmallFull smallFull = (SmallFull) request.getSession()
```

Comment:
A `SmallFullEntity` is stored in the session in the interceptor

Reply:
Good spot. I will refactor the name of the SmallFullEntity stored via the SmallFullInterceptor from SmallFull to SmallFullEntity I think that may have been responsible for the confusion.

Reply:
This should have failed during testing with a cast exception?